pkg/acquisition/modules/syslog/syslog.go

```diff
@@ -0,0 +1,210 @@
+	prometheus.CounterOpts{
+		Name: "cs_syslogsource_parsed_total",
+		Help: "Total lines that were successfully parsed",
+	},
+	[]string{"source", "type"})
+
+func (s *SyslogSource) GetName() string {
+	return "syslog"
+}
+
+func (s *SyslogSource) GetMode() string {
+	return s.config.Mode
+}
+
+func (s *SyslogSource) Dump() interface{} {
+	return s
+}
+
+func (s *SyslogSource) CanRun() error {
+	return nil
+}
+
+func (s *SyslogSource) GetMetrics() []prometheus.Collector {
+	return []prometheus.Collector{linesReceived, linesParsed}
+}
+
+func (s *SyslogSource) ConfigureByDSN(dsn string, labelType string, logger *log.Entry) error {
+	return fmt.Errorf("syslog datasource does not support one shot acquisition")
+}
+
+func (s *SyslogSource) OneShotAcquisition(out chan types.Event, t *tomb.Tomb) error {
+	return fmt.Errorf("syslog datasource does not support one shot acquisition")
+}
+
+func (s *SyslogSource) Configure(yamlConfig []byte, logger *log.Entry) error {
+	s.logger = logger
+	s.logger.Infof("Starting syslog datasource configuration")
+	syslogConfig := SyslogConfiguration{}
+	syslogConfig.Mode = configuration.TAIL_MODE
+	err := yaml.UnmarshalStrict(yamlConfig, &syslogConfig)
+	if err != nil {
+		return errors.Wrap(err, "Cannot parse syslog configuration")
+	}
+	if syslogConfig.Addr == "" {
+		syslogConfig.Addr = "127.0.0.1" //do we want a usable or secure default ?
+	}
+	if syslogConfig.Port == 0 {
+		syslogConfig.Port = 514
+	}
+	if syslogConfig.Proto == "" {
+		syslogConfig.Proto = "udp"
+	}
+	s.config = syslogConfig
+	return nil
+}
+
+func (s *SyslogSource) StreamingAcquisition(out chan types.Event, t *tomb.Tomb) error {
+	c := make(chan syslogserver.SyslogMessage)
+	s.server = &syslogserver.SyslogServer{}
+	s.server.SetChannel(c)
+	err := s.server.Listen(s.config.Addr, s.config.Port)
+	if err != nil {
+		return errors.Wrap(err, "could not start syslog server")
+	}
+	s.server.StartServer()
+	t.Go(func() error {
+		defer types.CatchPanic("crowdsec/acquis/syslog/live")
+		return s.handleSyslogMsg(out, t, c)
+	})
+	return nil
+}
+
+func (s *SyslogSource) buildLogFromSyslog(ts *time.Time, hostname *string,
+	appname *string, pid *string, msg *string) (string, error) {
+	ret := ""
+	if ts != nil {
+		ret += ts.Format("Jan 2 15:04:05")
+	} else {
+		ret += time.Now().Format("Jan 2 15:04:05")
+	}
+	if hostname != nil {
+		ret += " " + *hostname
+	} else {
+		ret += " unknownhost"
+	}
+	if appname != nil {
+		ret += " " + *appname
+	} else {
+		return "", errors.Errorf("missing appname field in syslog message")
+	}
+	if pid != nil {
+		/*
+			!!! ugly hack !!!
+			Due to a bug in the syslog parser we use (https://github.com/influxdata/go-syslog/issues/31),
+			the ProcID field will contain garbage if the message as a ] anywhere in it.
+			Assume that a correctly formated ProcID only contains number, and if this is not the case, set it to an arbitrary value
+		*/
+		_, err := strconv.Atoi(*pid)
+		if err != nil {
+			ret += "[4242]: "
```

Comment:
as an arbitrary value I would choose sth rather distinguishable like 0 (is it possible to have 0, or 1 as we all know that 1 is a very low probable value for this ?)

Reply:
Using 0 is not possible, as the grok pattern we use to parse a syslog line expects a positive integer in the PID field.